modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -172,3 +168,34 @@ const resolveNumericAggregation = (aggregations: NumericAggregations) => {
 	// TODO: implement
 	throw Error('Not implemented');
 };
+
+const emptyAggregation: Aggregations = { bucket_count: 0, buckets: [] };
+
+export class AggregationAccumulator {
+	requestedFields: string[];
+	totalAgg: RemoteAggregation;
+
+	constructor(requestedFieldsMap: RequestedFieldsMap) {
+		const requestedFields = Object.keys(requestedFieldsMap);
+		this.requestedFields = requestedFields;
+		/*
+		 * seed accumulator with the requested field keys
+		 * this will make it easier to add to using key lookup instead of Array.find
+		 */
+		this.totalAgg = requestedFields.reduce<RemoteAggregation>((accumulator: any, field: any) => {
```

Comment:
Wouldn't `field: string` work here?

Reply:
Added a generic, just forgot to remove the inline `any`.
The generic types these fields correctly once `any` is removed.